### DISCUSSION TITLE
[Proposal] [WIP] Achievement System

### DISCUSSION THREAD

**Discussion by 100gle** [laike9m/Python-Type-Challenges#65] [Ideas]
**the proposal description is WIP**

this proposal inspired by [Astro blog tutorial](https://docs.astro.build/en/tutorial/0-introduction/1/)

Now this project has many challenges. I think we can build a simple achievement system based on these items. 

My ideas are: 

- total scores are 100.
- different level has different weight. for example, the basic level is 0.1 and the extreme level is 0.4. So the finally score is weighted sum of each passed challenge's score.
- We can build a achievement card for user to show their achievement when finished all challenges. The card can be shared to social media.
- all achievements are stored in local storage. No any server-side database needed.

**Comment by laike9m** [comment]
Actually I have a TODO to track and show which challenges have been completed, and your proposal is one step further.

Overall the direction looks good. We can discuss more details like how to design achievement system.

**Personally, I'd like to start with something simple**, e.g. finished 1/5/10/etc challenge, finished all basic/intermediate/advanced challenges. The scoring system looks a bit complicated to me, especially considering we're constantly adding more challenges which may cause subtle bugs.

**Comment by laike9m** [comment]
I created an issue as the first step #78
